Allow kmod_t and udev_t confidentiality lockdown permission

zpytela opened the pull request:
The confidentiality permission from the lockdown class is required
for modprobe and systemd-udevd to use tracefs.

zpytela committed:
Add confidentiality lockdown permission into fs_rw_tracefs_files()

Since adding the lockdown class to selinux-policy, the confidentiality
permission starts to be checked on use of tracefs.
The permissions to access tracefs filesystem objects are a part of the
fs_rw_tracefs_files() and fs_manage_tracefs_dirs() interfaces, so the
confidentiality lockdown permission was added into these interfaces.

Example of services requiring this access are modprobe or systemd-udevd,
running in the kmod_t and udev_t domains. These domains are a part
of the unconfined_domain_type attribute, so the permission was added
for this attribute, too. Additionaly, explicit fs_rw_tracefs_files()
interface calls were added for kmod_t and udev_t.

WOnder93 commented:
Also here I would prefer to add it to `fs_rw_tracefs_files()` and possibly also to `fs_manage_tracefs_dirs()`. These domains currently don't require these interfaces, since they are part of `files[ystem]_unconfined_type`, but it can't hurt to be explicit and add the `fs_rw_tracefs_files()` call to them.

zpytela commented:
I see the point in adding it into `fs_rw_tracefs_files()`. It does not solve the problem for kmod or udev though, it needs to be added to I guess `devices_unconfined_type` - the new classes/capabilities are not allowed automatically.

zpytela commented:
I changed the commit to allow the permission to `devices_unconfined_type`: it is true it is only called from the `unconfined_domain` interface, but it seems to fit semantically here. I added it to the tracefs interfaces, too, and made kmod/udev call `fs_rw_tracefs_files()`.

WOnder93 commented:
Didn't you mean to add it to `filesystem_unconfined_type` instead? `/dev/kmem` and friends are in another PR (and there `integrity` is needed, not `confidentiality`.

Either way, I'd rather we add both permissions to `unconfined_domain_type` directly, since they cover various different things, not just device or filesystem related.

As a side note, given that `unconfined_domain_noaudit()` is the only caller of `fs_unconfined()` and `filesystem_unconfined()`, I wonder if we shouldn't remove the `files[ystem]_unconfined_type` attributes completely (granting the permissions directly in the interfaces) and just call the interfaces directly on `unconfined_domain_type`. Right now, the attributes are pretty much useless and just slow down access vector computation. (But this is of course for a separate PR, ideally starting at refpolicy...)

zpytela commented:
You are right, I worked on 2 commits at the same time, funny enough I noticed the commit message is incorrect, but not the commit content. Hope now it is consistent.

WOnder93 reviewed (approved):
Yes, it looks good now, thanks!
